frontend/src/metabase-lib/queries/structured/Filter.ts

```diff
@@ -61,6 +63,38 @@ export default class Filter extends MBQLClause {
     return this._query.removeFilter(this._index);
   }
 
+  betterDateLabel() {
+    const dimension = this.dimension();
+    const unit = dimension?.temporalUnit();
+    const op = this.operatorName();
+    const args = this.arguments();
+    const allStrs = args.every(arg => typeof arg === "string");
+    const isDate = ["day", "week", "month", "quarter", "year"].includes(unit);
```

Comment:
Here we only want to format day, week, month, quarter, and year, since I only added logic in place for those, falling back to the normal date labels otherwise.